src/legacy/ui/public/agg_types/buckets/histogram.js

```diff
@@ -158,31 +159,30 @@ export const histogramBucketAgg = new BucketAggType({
       }
     },
 
+    {
+      name: 'extend_bounds',
+      default: false,
+      editorComponent: ExtendBoundsParamEditor,
+      write: () => {},
+    },
+
     {
       name: 'extended_bounds',
-      default: {},
-      editor: extendedBoundsTemplate,
+      default: {
+        min: '',
+        max: '',
+      },
+      editorComponent: ExtendedBoundsParamEditor,
       write: function (aggConfig, output) {
-        const val = aggConfig.params.extended_bounds;
+        const { min, max } = aggConfig.params.extended_bounds;
 
-        if (aggConfig.params.min_doc_count && (val.min != null || val.max != null)) {
-          output.params.extended_bounds = {
-            min: val.min,
-            max: val.max
-          };
+        if (aggConfig.params.extend_bounds &&
+          (min || min === 0) &&
+          (max || max === 0)) {
+          output.params.extended_bounds = { min, max };
         }
       },
-
-      // called from the editor
-      shouldShow: function (aggConfig) {
-        const field = aggConfig.params.field;
-        if (
-          field
-          && (field.type === 'number' || field.type === 'date')
-        ) {
-          return aggConfig.params.min_doc_count;
-        }
-      }
+      shouldShow: aggConfig => aggConfig.params.extend_bounds
```

Comment:
The naming here is pretty confusing to me. There's `extend_bounds`, which is a boolean, and `extended_bounds`, which is a `{ min, max }` object? To resolve this, I would recommend naming booleans like `has_extended_bounds` or `is_extended`

Reply:
@maryia-lapata please, respond with your thoughts about better naming solution

Reply:
I agree that the name is a bit confusing. I'm fine with `has_extended_bounds`.

Reply:
Done